src/ServiceControl.Config/Commands/UpgradeAuditInstanceCommand.cs

```diff
@@ -0,0 +1,125 @@
+﻿namespace ServiceControl.Config.Commands
+{
+    using System;
+    using System.ServiceProcess;
+    using System.Threading.Tasks;
+    using Caliburn.Micro;
+    using Events;
+    using FluentValidation;
+    using Framework;
+    using Framework.Commands;
+    using Framework.Modules;
+    using ServiceControlInstaller.Engine.Configuration.ServiceControl;
+    using ServiceControlInstaller.Engine.Instances;
+    using ServiceControlInstaller.Engine.ReportCard;
+    using ServiceControlInstaller.Engine.Validation;
+    using UI.InstanceDetails;
+    using UI.MessageBox;
+    using UI.Upgrades;
+    using Validation;
+    using Xaml.Controls;
+    using Validations = Extensions.Validations;
+
+    class UpgradeAuditInstanceCommand : AwaitableAbstractCommand<InstanceDetailsViewModel>
+    {
+        readonly IEventAggregator eventAggregator;
+        readonly IWindowManagerEx windowManager;
+        readonly ServiceControlInstanceInstaller serviceControlInstaller;
+        readonly ServiceControlAuditInstanceInstaller serviceControlAuditInstaller;
+        readonly Func<string, AddNewAuditInstanceViewModel> auditUpgradeViewModelFactory;
+
+        [FeatureToggle(Feature.LicenseChecks)]
+        public bool LicenseChecks { get; set; }
+
+        public UpgradeAuditInstanceCommand(Func<InstanceDetailsViewModel, bool> canExecuteMethod = null) : base(canExecuteMethod)
+        {
+        }
+
+        public UpgradeAuditInstanceCommand(
+            IWindowManagerEx windowManager,
+            IEventAggregator eventAggregator,
+            ServiceControlInstanceInstaller serviceControlInstaller,
+            ServiceControlAuditInstanceInstaller serviceControlAuditInstaller,
+            Func<string, AddNewAuditInstanceViewModel> auditUpgradeViewModelFactory)
+        {
+            this.windowManager = windowManager;
+            this.eventAggregator = eventAggregator;
+            this.serviceControlInstaller = serviceControlInstaller;
+            this.serviceControlAuditInstaller = serviceControlAuditInstaller;
+            this.auditUpgradeViewModelFactory = auditUpgradeViewModelFactory;
+        }
+
+        public async override Task ExecuteAsync(InstanceDetailsViewModel model)
+        {
+            if (LicenseChecks)
```

Comment:
I cannot imagine that this code is already present somewhere else. Shouldn't the license check be invoked not as part of this method but prior to it?

Reply:
I agree consistency is more important